aep/general/0135/aep.md.j2

```diff
@@ -0,0 +1,217 @@
+# Standard methods: Delete
+
+In REST APIs, it is customary to make a `DELETE` request to a resource's URI
+(for example, `/v1/publishers/{publisher}/books/{book}`) in order to delete
+that resource.
+
+Resource-oriented design ([AEP-121][aep-121]) honors this pattern through the `Delete`
+method. This method accepts the URI representing that resource and usually returns
+an empty response.
+
+## Guidance
+
+APIs **should** generally provide a delete method for resources unless it is
+not valuable for users to do so.
+
+The Delete method **should** succeed if and only if a resource was present and
+was successfully deleted. If the resource did not exist, the method **should**
+send a `404 Not found` (`NOT_FOUND`) error.
+
+### Requests
+
+{% tab oas -%}
+
+Delete methods are specified using the following pattern:
+
+{% sample 'delete.oas.yaml', 'paths' %}
+
+- The HTTP verb **must** be `DELETE`.
+- There **must not** be a request body.
+  - If a delete request contains a body, the body **must** be ignored, and
+    **must not** cause an error.
+- The request **must not** require any fields in the query string. The request
+  **should not** include optional fields in the query string unless described
+  in another AEP.
+- Delete methods **must** return `204 No Content` with no response body.
+- If the API is operating on the [Management Plane][], the method should have
+  [strong consistency][]: the completion of a delete method **must** mean
+  that the existence of the resource has reached a steady-state and reading
+  resource state returns a consistent response.
+
+{% tab proto -%}
+
+Delete methods are specified using the following pattern:
+
+```proto
+rpc DeleteBook(DeleteBookRequest) returns (library.protobuf.Empty) {
+  option (library.api.http) = {
+    delete: "/v1/{name=publishers/*/books/*}"
+  };
+  option (library.api.method_signature) = "name";
+}
+```
+
+- The RPC's name **must** begin with the word `Delete`. The remainder of the
+  RPC name **should** be the singular form of the resource's message name.
+- The request message **must** match the RPC name, with a `Request` suffix.
+- The response message **should** be `library.protobuf.Empty`.
+  - If the delete RPC is [long-running](#long-running-delete), the response
+    message **must** be a `library.longrunning.Operation` which resolves to the
+    correct response.
+- The request message field receiving the resource name **should** map to the
```

Comment:
Because there was concern about duplicating guidance across the OAS and protobuf sections. Since this is already specified in the OAS section, it seems unnecessary here. Is there some good reason that it has to be here (duplicated)?

Reply:
Oh, I see.  I think we want this guidance for protobuf as well (for `google.api.http` annotations/JSON transcoding).  Maybe we factor this bullet point out of the IDL-specific sections?  That's the pattern I've been following on the other standard method adoptions.

Reply:
If we start on that road we will have to duplicate all the HTTP guidance in the protobuf section. Or we make the entirety of the HTTP guidance common. Neither of these seem particularly desirable.

Let's leave this as is for now and discuss amongst the group in our next meeting.

Reply:
Why wouldn't we make the HTTP guidance common?  The way I've been thinking about these tabs is that they're for IDL-specific guidance; using HTTP DELETE isn't specific to OAS.

Reply:
This is a good question -- is HTTP vs gRPC a distinction of protocols or of IDLs? I think of it more as protocols. HTTP must conform to the HTTP RFCs, gRPC not so. So if we made all the HTTP guidance common, then what does that mean for gRPC?

Reply:
My point is that the proto vs. OAS tabs _aren't_ gRPC vs. HTTP tabs, so HTTP guidance shouldn't be exclusive to the OAS tab - just OAS-specific guidance.

Reply:
Resolving so we can merge, but will send a follow-up PR where we can continue this conversation.